CRM/MembershipExtras/Test/Fabricator/PaymentPlan.php

```diff
@@ -0,0 +1,279 @@
+<?php
+use CRM_MembershipExtras_Test_Fabricator_RecurringContribution as RecurringContributionFabricator;
+use CRM_MembershipExtras_Test_Fabricator_RecurringLineItem as RecurringLineItemFabricator;
+use CRM_MembershipExtras_Test_Fabricator_LineItem as LineItemFabricator;
+use CRM_MembershipExtras_Test_Fabricator_Membership as MembershipFabricator;
+use CRM_MembershipExtras_Test_Fabricator_Contribution as ContributionFabricator;
+
+/**
+ * Class CRM_MembershipExtras_Test_Fabricator_PaymentPlan
+ */
+class CRM_MembershipExtras_Test_Fabricator_PaymentPlan {
+
+  /**
+   * Fabricates a payment plan with the provided information.
+   *
+   * @param array $recurringContributionParams
+   * @param array $lineItemParams
+   * @param array $contributionParams
+   *
+   * @return array
+   * @throws \Exception
+   */
+  public static function fabricate($recurringContributionParams, $lineItemParams, $contributionParams) {
```

Comment:
@omarabuhussein  @MiyaNoctem  I am not sure if it is a good idea but I think it is we should have default value for $recurringContributionParams, $lineItemParams, $contributionParams so when we fabricate the payment plan, these values are fabricated by default.    For example,  LineItem, Recurring Contribution, Contribution should be created by its own fabricator.

Reply:
as I replied above, when we usually create payment plan then we usally want control over certain paramterers, so I think it is fine if we don't have default values, and we can change that later if we discovered the need for it.

But I have another problem with the PaymentPlan fabricator, we usually and interchangeably refer to PaymentPlan and RecurContribution as if they are the same thing, but here we are also creating other stuff such as Memberships, so this is more like an Order, so I suggest to rename it to PaymentPlanOrder.